examples/foood/src/Editor.js

```diff
@@ -336,6 +361,21 @@ const RecipeEditor = ({
 
     const styles = useStyles();
 
+    const [warningModalOpen, setWarningModalOpen] = React.useState(false);
+    const [inMobileMode, setInMobileMode] = React.useState(false);
+    React.useEffect(() => {
+        const checkWindowSize = () => {
+            if (window.innerWidth < 600) {
+                setInMobileMode(true);
+            } else {
+                setInMobileMode(false);
+            }
+        };
+
+        checkWindowSize();
+        window.addEventListener('resize', checkWindowSize);
+    }, []);
```

Comment:
At the end of a `useEffect` where you've set up a listener, it's good practice to return a function that will unregister the listener. That function will get called when the component is unmounted from the DOM, and if the listener stays registered it can sometimes result in weird behavior. (in this case it would just result in react giving us some warnings in the console).
```suggestion
        return () => window.removeEventListener('resize', checkWindowSize);
    }, []);
```